core/src/avm1/activation.rs

```diff
@@ -2805,6 +2837,28 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         scope.define_local(name, value, self)
     }
 
+    /// Define a local property or set a value referenced by a target path string.
+    ///
+    /// If `name` is a path string, we search for ":" or ".". If we have one, we will resolve
+    /// as a target path, then if this activation operates in a local scope we will set the
+    /// variable on the mentioned object to `value`.
+    /// Otherwise, the behavior is the same as `define_local`.
+    pub fn define_local_or_set_variable(
+        &mut self,
+        name: AvmString<'gc>,
+        value: Value<'gc>,
+    ) -> Result<(), Error<'gc>> {
+        if name.find(b":.".as_ref()).is_some() {
+            if !self.in_local_scope() {
+                self.set_variable(name, value)
+            } else {
+                Ok(())
+            }
+        } else {
+            self.define_local(name, value)
```

Comment:
I think we could inline `define_local` here, and then rename this entire fn to `define_local`.